### ISSUE TITLE
Array fields for general data types in the Content Type Builder plugin

### ISSUE THREAD

**Issue by Aurelyn10** [strapi/strapi#1125]
**What is the expected behavior?**

How can I create multiple items (array) in the number or string fields in Content Type Builder plugin?
I want same key name/different value pairs, like as array or list.
I want to put multiple datas in one field like as current media field ( It has the option of “Allow multiple files”.)
I selected mongodb as database.
I already was suggested to create new plugin, but I am not expert to handle that.
So with current Content Type Builder, How can I implement following feature.
For example, Product Content has final sold price field with array value, and it is keeping update from client side’s API call. And this is related to Customer content, one Customer can have many Products, one Product can have many Customers. 
## Can I implement this database and API with this current Strapi CMS ?
I really want to use this CMS for implementing this feature. what is the solution?
Thank you.

---

- [ ] I'm sure that this issue hasn't already been referenced.

**Comment by enoy** [comment]
I also often need to create multiple items (array or list) in the number/string/text/media fields in Content Type Builder plugin. Is there a way to do this?

**Comment by raulriera** [comment]
> I also often need to create multiple items (array or list) in the number/string/text/media fields in Content Type Builder plugin. Is there a way to do this?

As a workaround, why not create a new contentType and set the relationship of one to many? A little bit of boilerplate, but doable today

**Comment by Aurelyn10** [comment]
I think it will increase complexity, if there is many items to be listed as array to even base data type such as string and number.

**Comment by Aurelsicoko** [comment]
You're right, currently, this kind of field (array) is complicated to handle with Strapi... I suggest voting for the **Plugin - Matrix/Repeater** [on the vote page](https://strapi.io/vote). Also, the JSON field could fix the issue in another way...

**Comment by lauriejim** [comment]
Related https://github.com/strapi/strapi/issues/784  https://github.com/strapi/strapi/pull/1313

**Comment by BertelsBenjamin** [comment]
I see this post dates back from 2018. However, the solution is to choose JSON and when filling in the actual data, just write an array like `["one", "two", "three"]` and eventually you can get the full json response by going to localhost:1337/products. You'll see that it returns an array.
Note: "/products" can be any name you gave to a content-type of course.

**Comment by congthang1** [comment]
@BertelsBenjamin how to edit this array like $pull or $push?

**Comment by PaulSinghDev** [comment]
I found the simplest solution to be to just have a `short text` field and add a regex validator `^[a-zA-Z0-9 ]+(?:,[a-zA-Z0-9 ]+)*$`. Then, when I fetch the result I can just `.split(',')` the field to have a proper array.
